src/reconcile/rec_visibility.c

```diff
@@ -715,6 +806,39 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, W
     /* Check the update chain for conditions that could prevent it's eviction. */
     WT_RET(__rec_validate_upd_chain(session, r, onpage_upd, select_tw, vpack));
 
+    /* Check if we need to delete any value from the history store. */
+    if (onpage_upd != NULL) {
+        for (delete_upd = tombstone != NULL ? tombstone : onpage_upd; delete_upd != NULL;
+             delete_upd = delete_upd->next) {
+            if (delete_upd->txnid == WT_TXN_ABORTED)
+                continue;
+
+            if (F_ISSET(delete_upd, WT_UPDATE_TO_DELETE_FROM_HS))
+                break;
+        }
+
+        if (delete_upd != NULL && delete_upd->type == WT_UPDATE_TOMBSTONE) {
+            delete_tombstone = delete_upd;
+            for (delete_upd = delete_tombstone->next; delete_upd != NULL;
```

Comment:
The `for, break, if, for` construct here seems odd to me. Is there a guarantee that there is only ever 0, 1 or 2 updates with WT_UPDATE_TO_DELETE_FROM_HS in an update chain, with 0 being the common case. 1 indicating that there is no tombstone (what does that mean?), and 2 indicating that there is a value and a tombstone?

Reply:
Yes, that's correct, at most two updates with the flag WT_UPDATE_TO_DELETE_FROM_HS. I'm not sure how this translate to the change of the code structure.

Reply:
Fixed in [ffea256](https://github.com/wiredtiger/wiredtiger/pull/7939/commits/ffea25641f7d62ac1919f643ba0d12ae1b3c2075)